nav2_theta_star_planner/include/nav2_theta_star_planner/theta_star.hpp

```diff
@@ -0,0 +1,244 @@
+   * @param curr_id sends in the index of the goal coordinate, as stored in nodes_position
+   */
+  void backtrace(std::vector<coordsW> & raw_points, int curr_id);
+
+  /**
+   * @brief performs the line of sight check using Bresenham's Algorithm,
+   *            and has been modified to calculate the traversal cost incurred in a straight line path between
+   *            the two points whose coordinates are (x0, y0) and (x1, y1)
+   * @param sl_cost is used to return the cost thus incurred
+   * @return true if a line of sight exists between the points
+   */
+  bool losCheck(
+    const int & x0, const int & y0, const int & x1, const int & y1,
+    double & sl_cost);
+
+  void initializePosn(int size_inc = 0);
+
+  void setContainers();
+
+  double dist(const int & ax, const int & ay, const int & bx, const int & by)
+  {
+    return std::hypot(ax - bx, ay - by);
+  }
+
+  double getEuclideanCost(const int & ax, const int & ay, const int & bx, const int & by)
+  {
+    return w_euc_cost_ * dist(ax, ay, bx, by);
+  }
+
+  double getCellCost(const int & cx, const int & cy) const
+  {
+    return /*50 + 0.8 * */ costmap_->getCost(cx, cy);
+  }
+
+  /**
+   * @brief for the point(cx, cy) its traversal cost is calculated by <parameter>*(<actual_traversal_cost_from_costmap>)^2/(<max_cost>)^2
+   * @return the traversal cost thus calculated
+   */
+  double getTraversalCost(const int & cx, const int & cy)
+  {
+    double curr_cost = getCellCost(cx, cy);
+    return w_traversal_cost_ * curr_cost * curr_cost / LETHAL_COST / LETHAL_COST;
+  }
+
+  double getHCost(const int & mx, const int & my)
+  {
+    return w_heuristic_cost_ * dist(mx, my, dst_.x, dst_.y);
+  }
+
+  /**
+   * @brief it is an overloaded function to ease in cost calculations while performing the LOS check
+   * @param cx
+   * @param cy
+   * @param cost denotes the total straight line traversal cost, adds the traversal cost for the node (cx, cy) at every instance
+   */
+  bool isSafe(const int & cx, const int & cy, double & cost) const
+  {
+    double curr_cost = getCellCost(cx, cy);
+    if (curr_cost < LETHAL_COST) {
+      cost += w_traversal_cost_ * curr_cost * curr_cost / LETHAL_COST / LETHAL_COST;
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  void addIndex(const int & cx, const int & cy, const int & id_this)
+  {
+	node_position_[size_x_ * cy + cx] = id_this;
+  }
+
+  void getIndex(const int & cx, const int & cy, int & id_this)
+  {
+    id_this = node_position_[size_x_ * cy + cx];
+  }
+
+  bool withinLimits(const int & cx, const int & cy) const
+  {
+    return cx >= 0 && cx < size_x_ && cy >= 0 && cy < size_y_;
+  }
+
+  bool isGoal(const tree_node & this_node) const
+  {
+    return this_node.x == dst_.x && this_node.y == dst_.y;
+  }
+
+  void addToNodesData(const int & id_this)
+  {
+    if (nodes_data_.size() <= static_cast<unsigned int>(id_this)) {
+      nodes_data_.push_back({});
+    } else {
+	  nodes_data_[id_this] = {};
```

Comment:
Indentation here is weird - maybe a tab?

Reply:
I also don't totally understand `addToNodesData` this function. What's going on here?